### PR TITLE
Cleanup xticks/yticks docstrings.

### PR DESCRIPTION
**Pull request by anntzer**
- Don't bother listing call signatures (the normal function signature is
  fine).
- Slightly tighten the parameters/return value descriptions.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by timhoffm** [comment]
This will only be correct after #15788 is in. For the current state, there would be no hint thatthat labels without ticks don't work.

Also IMHO the MATLAB-like API should somehow be mentioned (that was part of the pupose of the call signature). Having the same function work as getter/setter depending on passing parameters is a bit unusual for Python. To be precise, while the return value is documented, it is not obvious that `plt.xticks()` the the canonical way of getting the ticks and labels.

**Comment by anntzer** [comment]
The beginning of the docstring literally states "Get or set ...".  I guess "Set and get" would be slightly more accurate?

**Comment by timhoffm** [comment]
> I guess "Set and get" would be slightly more accurate?

Technically, yes, but with the inexplicit "Set nothing and get", which is quite obscure.

**Comment by anntzer** [comment]
"Set and/or get"?

**Comment by timhoffm** [comment]
Formally correct, but ugly (which reflects the API). I'm +0.5 for just sticking with "Get or set" plus an extra sentence "Pass no argument to return the current value without modifying it."

**Comment by anntzer** [comment]
sounds good to me, done

File: lib/matplotlib/pyplot.py [hunk 1]
```diff
@@ -1371,30 +1371,23 @@ def ylim(*args, **kwargs):
...
-    ticks : array-like
-        A list of positions at which ticks should be placed. You can pass an
-        empty list to disable xticks.
-
+    ticks : array-like, optional
+        The list of xtick locations.  Passing an empty list removes all xticks.
     labels : array-like, optional
-        A list of explicit labels to place at the given *locs*.
-
+        The labels to place at the given *locs*.
```
Comment (dstansby) [inline lib/matplotlib/pyplot.py]: It's not clear to me what `*locs*` is here - should it be the `*ticks*` kwarg instead?

Comment (anntzer) [reply lib/matplotlib/pyplot.py]: fair point, clarified.

**Comment by anntzer** [comment]
Also compressed the examples a bit.

File: lib/matplotlib/pyplot.py [hunk 2]
```diff
@@ -1371,30 +1371,23 @@ def ylim(*args, **kwargs):
 def xticks(ticks=None, labels=None, **kwargs):
     """
     Get or set the current tick locations and labels of the x-axis.
-
-    Call signatures::
-
-        locs, labels = xticks()            # Get locations and labels
-        xticks(ticks, [labels], **kwargs)  # Set locations and labels
+    Pass no arguments to return the current values without modifying them.
```
Comment (timhoffm) [inline lib/matplotlib/pyplot.py]: Needs an empty line before (Docstring should have a one-line sentence/summary. Everything else should be separate paragraphs).

File: lib/matplotlib/pyplot.py [hunk 3]
```diff
@@ -1446,30 +1426,23 @@ def xticks(ticks=None, labels=None, **kwargs):
 def yticks(ticks=None, labels=None, **kwargs):
     """
     Get or set the current tick locations and labels of the y-axis.
-
-    Call signatures::
-
-        locs, labels = yticks()            # Get locations and labels
-        yticks(ticks, [labels], **kwargs)  # Set locations and labels
+    Pass no arguments to return the current values without modifying them.
```
Comment (timhoffm) [inline lib/matplotlib/pyplot.py]: As above, empty line before.

File: lib/matplotlib/pyplot.py (hunk 2 above)
Comment (anntzer) [reply lib/matplotlib/pyplot.py]: done

File: lib/matplotlib/pyplot.py (hunk 3 above)
Comment (anntzer) [reply lib/matplotlib/pyplot.py]: done

**Comment by anntzer** [comment]
happy to wait for the discussion on the other PR to be resolved first

**Comment by anntzer** [comment]
Edited as needed following the rejection of #15788.

**Pull request merged**